pkg/bmc/bmc.go

```diff
@@ -347,3 +435,32 @@ func redfishGetSystemSecureBoot(redfishClient *gofish.APIClient, systemIndex int
 
 	return sboot, nil
 }
+
+// redfishGetPowerControl gets the specified PowerControl from the first chassis with a power link from the redfish API.
+func redfishGetPowerControl(
+	redfishClient *gofish.APIClient, powerControlIndex int) (*redfish.PowerControl, error) {
+	chassisCollection, err := redfishClient.GetService().Chassis()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chassis collection: %w", err)
+	}
+
+	for chassisIndex, chassis := range chassisCollection {
+		power, err := chassis.Power()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get power for chassis index %d: %w", chassisIndex, err)
+		}
+
+		if power == nil {
```

Comment:
I'm curious: is this just overly defensive programming or is it possible having nil pointers for some of the elements of this slice?

Reply:
Yes, `chassis.Power()` returns `nil, nil` if there is no link to power for that chassis, I already got hit with a nil pointer error for that :)

Ideally you should be able to specify which chassis to get the power from, but we can't use an index since I found out the hard way that gofish fetches all the chassis concurrently so they're not in the same order the api returns them.

This is just the method I went with since the only chassis with a power link is the one that I want for my use cases, but it's not as robust as I'd like

Reply:
I see, ok. So I didn't know that about the concurrent requests... gofish uses the same mechanism for every collection, so using that index is probably a bad idea :(

Reply:
For `PowerControl` in this PR it'll be fine since they're not links, but yes we probably need a better way to identify systems. Maybe we store an optional `systemId` and default to just using the first system returned otherwise? I expect most of my use cases to be only one system per BMC but good to leave the option for multiple systems open

Reply:
The checking for len of power.PowerControl is just making sure the index is not out of bounds. I kept the form from `redfishGetSystem` for symmetry but I think it would be clearer to write `powerControlIndex >= len(power.PowerControl)` as the condition.

But the iteration is to check for the first chassis that has the power link, since it there may be multiple enclosures that don't have the power link. It's definitely not ideal though, since I really want the chassis that has the same Id as the system Id

Reply:
I'm ok with that as a workaround, but we should definitely consider/explore other options to improve this in further PR/s. I can investigate this.

Reply:
Thanks for investigating, I'm open to changing this in the future!